google-cloud-storage/acceptance/storage/bucket_generate_signed_post_policy_v4_test.rb

```diff
@@ -72,9 +72,9 @@
     # Only defined when using a service account
     issuer = iam_client.authorization.issuer
     signer = lambda do |string_to_sign|
-      request = {
-        "payload": string_to_sign,
-      }
+      request = Google::Apis::IamcredentialsV1::SignBlobRequest.new(
```

Comment:
What happened here?

Reply:
@frankyn It appears that using the `Google::Apis::IamcredentialsV1::SignBlobRequest` request object instead of the hash resolves both the new `ArgumentError: unknown keyword: payload` problem and avoids the `` error. I'm guessing you were right about `payload :` and `payload =>`? causing a gotcha / issue with the json body between the two. I can still log the requests if you are interested.

Reply:
that's wild. this wasn't required before, but I'm not opposed to this change. If it's not too much trouble could you log request bodies for both?

Reply:
I [posted the logs](https://github.com/googleapis/google-cloud-ruby/issues/11224#issuecomment-829345579) to #11224.